crates/block-producer/src/mempool/dependency_graph.rs

```diff
@@ -280,7 +307,17 @@ impl<K: Ord + Copy, V: Clone> DependencyGraph<K, V> {
     ///
     /// This method assumes the node exists. Caller is responsible for ensuring this is true.
     fn try_make_root(&mut self, key: K) {
-        debug_assert!(self.vertices.contains_key(&key), "Potential root must exist in the graph");
+        if self.pending.contains(&key) {
+            return;
+        }
```

Comment:
Why is it not an error to try to make a pending node into a root?

Reply:
This method is private and called from a few locations. Effectively either callers have to perform this check, or this method could enforce it. I thought it safer to do it here in one place.

More specifically, this is called when

1. A pending node is promoted (so this check isn't necessary)
2. A root node is processed, so we check all of its children. We would have to filter children here.
3. Subgraphs are reverted, where we have to check and filter a bunch of nodes